### DISCUSSION TITLE
Question / Idea - spawning the repl with a pre-established connection.

### DISCUSSION THREAD

**Discussion by codefriar** [jsforce/jsforce#1583] [Ideas]
Greetings,

First, let me say, I love your work. 

I'd like to ask if you have any guidance on how to spawn a JSForce repl from within a typescript project. Specifically, I'd like to fire up a repl from within a sf extension. The idea being I can call `sf repl -o MyOrgAlias` and be dropped into an authorized repl session. 

I attempted this, apparently hamfistedly, with this code: 

```ts
import { Flags, SfCommand } from '@salesforce/sf-plugins-core';
import { AuthInfo, Connection, Messages, Org } from '@salesforce/core';
import { Cli } from 'jsforce/lib/cli/cli.js';
import { Connection as Conn } from 'jsforce';

// ... some boilerplate removed for brevity

export default class BahRepl extends SfCommand<BahReplResult> {
  // ... some boilerplate removed for brevity
  // flags definition hidden for brevity

  public async run(): Promise<BahReplResult> {
    const { flags } = await this.parse(BahRepl);
   // ------ The meat of it ------- ///
    const cli = new Cli();
    const connection: Connection =
    await this.getConnection(flags['target-org']);
    // eslint-disable-next-line no-underscore-dangle
    cli._conn = new Conn({
      instanceUrl: connection.instanceUrl,
      accessToken: connection.accessToken ?? '',
    });
    cli.saveCurrentConnection();

    await cli.start();

    return {
      path: 'src/commands/bah/repl.ts',
    };
  }
  // eslint-disable-next-line class-methods-use-this
  private async getConnection(targetOrg: Org): Promise<Connection> {
    const authInfo = await AuthInfo.create({ username: targetOrg.getUsername() });
    return Connection.create({ authInfo });
  }
}
```

Unfortunately, this does not actually give me an authorized repl. 

If, it's not currently possible to programatically spawn a repl with a pre-configured authorized session, my question then becomes ... can I have one? 

Use case: It's effectively a cli workbench replacement. very handy for prototyping things.

**Comment by cristiand391** [comment]
hey, sorry for the late reply!

our intern build this a few days ago, here's the base command:
https://github.com/soridalac/sf-repl-plugin/blob/main/src/commands/repl.ts
